File: Tree/BST.c
#include<stdio.h>
#include<stdlib.h>

struct node {
    int key;
    struct node *right, *left;

};

// creating a node:
struct node *newNode(int item){
    struct node *temp = (struct node*)malloc(sizeof(struct node));
    temp->key = item;
    temp->left = temp->right = NULL;
    return temp;
};

// insertion
struct node *insert( struct node *node, int key){
    if (node == NULL){
        return newNode(key);
    }

    if (key < node->key){
        node->left = insert(node->left,key);
    }else{
        node->right = insert(node->right,key);
    }
    return node;
};

//inorder

void inorder(struct node *root) {
  if (root != NULL) {
    // Traverse left
    inorder(root->left);

    // Traverse root
    printf("%d -> ", root->key);

    // Traverse right
    inorder(root->right);
  }
}

int main(){
    struct node *root = NULL;

    root = insert(root,8);
    root = insert(root,10);
    root = insert(root,2);
    inorder(root);
}

File: Tree/BST.py
#Binary Search Tree

#creating node
class Node:
    def __init__(self,key):
        self.key = key
        self.right = None
        self.left = None


# Inorder Traverdsal
def inOrder(root):
    if root is not None:
        inOrder(root.left)

        print(str(root.key) + "->",end=' ')

        inOrder(root.right)

#insertion
def insert(node,key):
    if node is None:
        return Node(key)
    if key<node.key:
        node.left = insert(node.left,key)
    else:
        node.right = insert(node.right,key)

    return node


root = None
root = insert(root,8)
root = insert(root,5)
root = insert(root,9)
root = insert(root,1)
inOrder(root)